addons/website/models/website_visitor.py

```diff
@@ -279,9 +279,9 @@ def _get_visitor_from_request(self, force_create=False, force_track_values=None)
 
         visitor = self.env['website.visitor'].sudo().search([('access_token', '=', access_token)])
 
-        if not force_create and not self.env.cr.readonly and visitor and not visitor.timezone:
+        if not force_create and not self.env.cr.readonly and visitor:
             tz = self._get_visitor_timezone()
-            if tz:
+            if tz != visitor.timezone:
                 visitor._update_visitor_timezone(tz)
```

Comment:
As we discussed, I am not sure it is worth the change. This method is called in a lot of places so I would avoid updating the timezone / get visitor timezone to handle corner cases